svm/src/account_saver.rs

```diff
@@ -42,15 +41,13 @@ fn max_number_of_accounts_to_collect(
 
 pub fn collect_accounts_to_store<'a, T: SVMMessage>(
     txs: &'a [T],
-    processing_results: &'a mut [TransactionProcessingResult],
-    durable_nonce: &DurableNonce,
-    lamports_per_signature: u64,
+    processing_results: &'a [TransactionProcessingResult],
 ) -> (Vec<(&'a Pubkey, &'a AccountSharedData)>, Vec<Option<&'a T>>) {
     let collect_capacity = max_number_of_accounts_to_collect(txs, processing_results);
     let mut accounts = Vec::with_capacity(collect_capacity);
     let mut transactions = Vec::with_capacity(collect_capacity);
-    for (processing_result, transaction) in processing_results.iter_mut().zip(txs) {
-        let Some(processed_tx) = processing_result.processed_transaction_mut() else {
```

Comment:
Do you mind removing `processed_transaction_mut` from `TransactionProcessingResultExtensions` now that it's no longer used?